### PR TITLE
Update shell scripts documentation

### PR DESCRIPTION
**Pull request by elParaguayo**
It seems people are unclear how to use `cmd-obj`. This commit adds more detail to the documentation and provides some better error handling where object identifiers are missing.

Would appreciate a review of this to make sure I've understood it correctly.

I've also edited the documentation for `qtile shell` and `qtile top`

### CODE REVIEW

**Review by ramnes — APPROVED** [review]
LGTM

File: docs/manual/commands/shell/qtile-cmd.rst [hunk 1]
```diff
@@ -4,6 +4,43 @@ qtile cmd-obj
...
 This can be used standalone or in other shell scripts.
 
+How it works
+------------
+
+``qtile cmd-obj`` works by selecting a command object and calling a specified function of that object.
+
+As per :ref:`commands-api`, Qtile's object graph has seven nodes: ``layout``, ``window``, ``group``,
+``bar``, ``widget``, ``screen``, and a special ``root`` node. These are the objects that can be accessed
+via ``qtile cmd-obj`` (NB the root node is called ``cmd`` when using the script).
```
Comment (flacjacket) [inline docs/manual/commands/shell/qtile-cmd.rst:14]:
```suggestion
via ``qtile cmd-obj`` (NB the root node is called ``cmd`` when using the ``cmd-obj`` script to give it an addressable name).
```

File: docs/manual/commands/shell/qtile-cmd.rst [hunk 2]
```diff
@@ -4,6 +4,43 @@ qtile cmd-obj
...
+
+Running the command without a function (``-f``) will run the ``help`` command and list the commands
+available to the selected object. Commands shown with an asterisk ("*") require arguments to be passed via
```
Comment (flacjacket) [inline docs/manual/commands/shell/qtile-cmd.rst:17]:
```suggestion
Running the command against a selected object without a function (``-f``) will run the ``help`` command and list the commands
available to the object. Commands shown with an asterisk ("*") require arguments to be passed via
```

File: docs/manual/commands/shell/qtile-cmd.rst [hunk 3]
```diff
@@ -4,6 +4,43 @@ qtile cmd-obj
...
+
+Selecting an object
+~~~~~~~~~~~~~~~~~~~
+
+With the exception of ``cmd``, all objects need an identifier so the correct object can be selected. Refer to
+:ref:`object_graph_keys` for more information.
+
+.. note::
+
+    You will see from the graph on :ref:`commands-api` that certain objects can be accessed from other object.
```
Comment (flacjacket) [inline docs/manual/commands/shell/qtile-cmd.rst:28]:
```suggestion
    You will see from the graph on :ref:`commands-api` that certain objects can be accessed from other objects.
```

File: docs/manual/commands/shell/qtile-cmd.rst [hunk 4]
```diff
@@ -4,6 +4,43 @@ qtile cmd-obj
...
+    For example, ``qtile cmd-obj -o group term layout`` will list the commands for the current layout on the
+    ``term`` group.
+
+Information on functions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Running a function with the ``-i`` flag will provide additional detail about that function (i.e. what it does and what
+arguments it expects).
+
+.. warning::
```
Comment (flacjacket) [inline docs/manual/commands/shell/qtile-cmd.rst:38]: Maybe put this in a separate `Passing arguments to functions` section?

**Review by flacjacket — APPROVED** [review]
Having this extra documentation is great! I added a couple suggestions, but overall looks good.

**Comment by elParaguayo** [comment]
Thanks. All good suggestions. Updated commit should cover all of these.

File: docs/manual/commands/shell/qshell.rst [hunk 5]
```diff
@@ -23,21 +23,38 @@ builtin "cd" and "ls" commands act like their familiar shell counterparts:
     > ls
     layout/  widget/  screen/  bar/     window/  group/
 
-    > cd bar
+    > cd screen
+    layout/  window/  bar/  widget/
```
Comment (elParaguayo) [inline docs/manual/commands/shell/qshell.rst:27]: This required #2688 to be merged as that adds the ability to access `widget` from the `screen`.

**Pull request merged**